From 383202d8d4feb8cd0c1e3872c99456ffb2d6f2cf Mon Sep 17 00:00:00 2001
From: xxm <xxm@rock-chips.com>
Date: Fri, 8 Aug 2014 11:13:52 +0800
Subject: [PATCH] rk: fb: 1.new intterupt set interface 	2.use macro from
 rockchip-iovmm.h instead of string to refer mmu compatible name

---
 drivers/video/rockchip/lcdc/rk3288_lcdc.c | 5 +++--
 drivers/video/rockchip/rk_fb.c            | 5 ++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/video/rockchip/lcdc/rk3288_lcdc.c b/drivers/video/rockchip/lcdc/rk3288_lcdc.c
index 7d6b9a210e0d..0f5b6c195690 100755
--- a/drivers/video/rockchip/lcdc/rk3288_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk3288_lcdc.c
@@ -26,6 +26,7 @@
 #include <linux/interrupt.h>
 #include <linux/platform_device.h>
 #include <linux/clk.h>
+#include <linux/rockchip-iovmm.h>
 #include <asm/div64.h>
 #include <asm/uaccess.h>
 #include <linux/rockchip/cpu.h>
@@ -3611,9 +3612,9 @@ static int rk3288_lcdc_probe(struct platform_device *pdev)
 
 	if (dev_drv->iommu_enabled) {
 		if(lcdc_dev->id == 0){
-			strcpy(dev_drv->mmu_dts_name, "iommu,vopb_mmu");
+			strcpy(dev_drv->mmu_dts_name, VOPB_IOMMU_COMPATIBLE_NAME);
 		}else{
-			strcpy(dev_drv->mmu_dts_name, "iommu,vopl_mmu");
+			strcpy(dev_drv->mmu_dts_name, VOPL_IOMMU_COMPATIBLE_NAME);
 		}
 	}
 
diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index d1bee87f6200..22f9134ca121 100755
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -49,8 +49,7 @@
 
 #if defined(CONFIG_ION_ROCKCHIP)
 #include <linux/rockchip_ion.h>
-#include <linux/rockchip/iovmm.h>
-#include <linux/rockchip/sysmmu.h>
+#include <linux/rockchip-iovmm.h>
 #include <linux/dma-buf.h>
 #include <linux/highmem.h>
 #endif
@@ -3668,7 +3667,7 @@ int rk_fb_register(struct rk_lcdc_driver *dev_drv,
 			    rk_fb_get_sysmmu_device_by_compatible(dev_drv->mmu_dts_name);
 			if (mmu_dev) {
 				rk_fb_platform_set_sysmmu(mmu_dev, dev_drv->dev);
-				rockchip_sysmmu_set_fault_handler(dev_drv->dev,
+				rockchip_iovmm_set_fault_handler(dev_drv->dev,
 								  rk_fb_sysmmu_fault_handler);
 				rockchip_iovmm_activate(dev_drv->dev);
 			} else
-- 
2.35.3

